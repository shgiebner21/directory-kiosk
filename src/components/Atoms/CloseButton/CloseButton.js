import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Button } from 'semantic-ui-react'

export default class CloseButton extends Component {
  static propTypes = {
    /** Display a popup with the provided message */
    size: PropTypes.string,
    /** Content to display in button */
    content: PropTypes.string,
    /** Function called when the button is clicked */
    onClick: PropTypes.func,
  }
  static defaultProps = {
    size: 'small',
    content: 'Close'
  }

  render() {
    const { size, content, onClick, ...rest } = this.props

      return (
        <Button
          color="red"
          content={content}
          inverted
          icon="delete"
          size={size}
          onClick={onClick}
          data-testid="close-button"
          {...rest}
        />
      )

  }
}
