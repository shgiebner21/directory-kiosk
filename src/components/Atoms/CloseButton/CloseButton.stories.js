import React from 'react'

import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import centered from '@storybook/addon-centered'
import { withKnobs, select, text } from '@storybook/addon-knobs'

import CloseButton from './CloseButton'

const stories = storiesOf('Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('CloseButton', 
withInfo()(() => {
  const sizeLabel = 'Size'
  const options = {
    mini: 'mini',
    tiny: 'tiny',
    small: 'small',
    medium: 'medium',
    large: 'large',
    big: 'big',
    huge: 'huge',
    massive: 'massive',
  }
  const defaultValue = 'medium'
  const value = select(sizeLabel, options, defaultValue)

  const textLabel = 'Content'
  const textDefaultValue = 'Close Me'
  const textValue = text(textLabel, textDefaultValue)

  return (
    <CloseButton
      size={value}
      content={textValue}
      onClick={() => {
        alert("Agggggghhh, I'm melting!")
      }}
    />
  )
})
)
