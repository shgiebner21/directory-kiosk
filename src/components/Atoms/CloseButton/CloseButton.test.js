import React from 'react'

import CloseButton from './CloseButton'

describe('CloseButton', () => {
  const buttonLabel = 'Click me'
  const mockFun = jest.fn()

  it('CloseButton matches the snapshot', () => {
    const wrapper = window.shallow(<CloseButton onClick={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })


  it('Uses the passed onClick function', () => {
    const wrapper = window.shallow(<CloseButton onClick={mockFun} />)
    wrapper.simulate('click')
    expect(mockFun).toHaveBeenCalled()
  })
})
