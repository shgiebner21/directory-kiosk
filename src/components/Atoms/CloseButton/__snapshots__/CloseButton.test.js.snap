// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton CloseButton matches the snapshot 1`] = `ShallowWrapper {}`;
